app/src/main/java/ch/epfl/polychef/fragments/UserProfileFragment.java

```diff
@@ -0,0 +1,123 @@
+package ch.epfl.polychef.fragments;
+
+import android.content.Context;
+import android.os.Bundle;
+import android.view.LayoutInflater;
+import android.view.View;
+import android.view.ViewGroup;
+import android.widget.TextView;
+
+import androidx.annotation.NonNull;
+import androidx.annotation.Nullable;
+import androidx.core.view.ViewCompat;
+import androidx.fragment.app.Fragment;
+import androidx.recyclerview.widget.LinearLayoutManager;
+import androidx.recyclerview.widget.RecyclerView;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import ch.epfl.polychef.CallHandler;
+import ch.epfl.polychef.R;
+import ch.epfl.polychef.pages.HomePage;
+import ch.epfl.polychef.recipe.Recipe;
+import ch.epfl.polychef.users.User;
+import ch.epfl.polychef.utils.RecipeMiniatureAdapter;
+
+
+public class UserProfileFragment extends Fragment implements CallHandler<Recipe> {
+
+    public UserProfileFragment() {
+        // Required empty public constructor
+    }
+
+    private HomePage hostActivity;  //TODO use ConnectedActivity if possible
```

Comment:
Concerning the comment: HomePage already `extends ConnectedActivity` thus every methods of ConnectedActivity is usable through the hostActivity (if that was your problem)

Reply:
I think the idea was to move some methods from HomePage to ConnectedActivity (e.g. get*Storage()) and only then use ConnectedActivity everywhere.

Reply:
Okay, so it can be done later, all good for me